packages/protobuf-test/README.md

```diff
@@ -10,6 +10,69 @@ Many test cases are run several times, once with the generated TypeScript code,
 once with the generated JavaScript code, and once with types created at runtime 
 from a file descriptor set.
 
-Two cover the code path for our string-based fallback for 64-bit integers, the
+To cover the code path for our string-based fallback for 64-bit integers, the
 entire suite should be run with the environment variable 
 `BUF_BIGINT_DISABLE=1`, which disables our BigInt feature detection. 
+
+## TypeScript
+
+Our TypeScript compatibility tests are run by the Make target `test-ts-compat`.
+Running this target will iterate over all versions specified in a sorted, 
+pre-configured list in the Makefile and use that version to compile protobuf-es.  
+If any compilation error is encountered, the process will exit immediately.
+
+The versions list is compiled of:
+
+- the earliest TypeScript version we support (4.1.2).
+- the earliest version we support with `skipLibChecks` set to `false`.
+- the current release candidate
+- the latest patch release of all minor versions in between
+
+### Adding a new version
+
+The following instructions will illustrate how to add a new version to the 
+compatibility tests.  
+
+For example, assuming we are adding version `4.5.3`.  
+
+- Run the following command from the project root.  This will install the 
+specified version of TypeScript as a package alias to `protobuf-test`.  Note 
+that the version before the `@` sign is the version number with all decimals 
+replaced with underscores:
+
+```bash
+  npm i -w packages/protobuf-test ts4_5_3@npm:typescript@4.5.3 --save-exact
+```
```

Comment:
You can indent code block, but any empty lines in between need to indented too:

```suggestion
  
  ```bash
  npm i -w packages/protobuf-test ts4_5_3@npm:typescript@4.5.3 --save-exact
  ```
```

Note that line 42 has two spaces.